edx_django_utils/data_generation/management/commands/manufacture_data.py

```diff
@@ -0,0 +1,328 @@
+            body = f"PK: {self.instance.pk}"
+        else:
+            body = f"fields: {self.customizations}"
+        ret = ("\t" * level) + f"{repr(self.data)} {body}" + "\n"
+        for child in self.children:
+            ret += child.__str__(level + 1)
+        return ret
+
+    def __repr__(self):
+        """
+        Overridden repr
+        """
+        return f'<Tree Node {self.data}>'
+
+
+def build_tree_from_field_list(list_of_fields, provided_factory, base_node, customization_value):
+    """
+    Builds a non-binary tree of nodes based on a list of children nodes, using a base node and it's associated data
+    factory as the parent node the user provided value as a reference to a potential, existing record.
+
+    - list_of_fields (list of strings): the linked list of associated objects to create. Example-
+        ['enterprise_customer_user', 'enterprise_customer', 'site']
+    - provided_factory (factory.django.DjangoModelFactory): The data factory of the base_node.
+    - base_node (Node): The parent node of the desired tree to build.
+    - customization_value (string): The value to be assigned to the object associated with the last value in the
+        ``list_of_fields`` param. Can either be a FK if the last value is a subfactory, or alternatively
+        a custom value to be assigned to the field. Example-
+        list_of_fields = ['enterprise_customer_user', 'enterprise_customer', 'site'],
+        customization_value = 9
+        or
+        list_of_fields = ['enterprise_customer_user', 'enterprise_customer', 'name'],
+        customization_value = "FRED"
+    """
+    current_factory = provided_factory
+    current_node = base_node
+    for index, value in enumerate(list_of_fields):
+        try:
+            # First we need to figure out if the current field is a sub factory or not
+            f = getattr(current_factory, value)
+            if isinstance(f, SubFactory):
+                fk_object = None
+                f_model = f.get_factory()._meta.get_model_class()
+
+                # if we're at the end of the list
+                if index == len(list_of_fields) - 1:
+                    # verify that the provided customization value is a valid pk for the model
+                    try:
+                        fk_object = f_model.objects.get(pk=customization_value)
+                    except f_model.DoesNotExist as exc:
+                        raise CommandError(
+                            f"Provided FK value: {customization_value} does not exist on {f_model.__name__}"
+                        ) from exc
+
+                # Look for the node in the tree
+                if node := current_node.find_value(f_model.__name__):
+                    # Not supporting customizations and FK's
+                    if (bool(node.customizations) or bool(node.children)) and bool(fk_object):
+                        raise CommandError("This script does not support customizing provided existing objects")
```

Comment:
to test this code, you have to specify a sub-factory modification, then specify a subobject with a PK. Something like:
```
    def test_command_cannot_customize_subfactories_of_existing_objects(self):
        """
        Test something idk
        """
        assert TestPerson.objects.all().count() == 0
        assert TestContactInfo.objects.all().count() == 0
        joe = TestPersonFactory()

        created_object = call_command(
            self.command,
            model="edx_django_utils.data_generation.tests.models.TestPersonContactPhoneNumber",
            field_customizations={
                # "phone_number": "(000) 000-0000",
                "test_contact_info__test_person__first_name": "shmoe",    <== we request a contact info's person's name 
                "test_contact_info__test_person": joe.email,   <== we then ask for a specific person for the contact info
            },
        )
```

Reply:
the comment in the code could maybe be a bit clearer

Reply:
I should mention the above code assumes email is the primary key of the person table